src/main/java/net/dries007/tfc/objects/recipes/UnmoldRecipe.java

```diff
@@ -5,182 +5,172 @@
+    public UnmoldRecipe(ResourceLocation group, NonNullList<Ingredient> input, @Nonnull Metal.ItemType type, float chance)
     {
-        this.mold = mold;
+        super(group, input, ItemStack.EMPTY);
+        this.type = type;
+        this.chance = chance;
     }
 
     @Override
-    public boolean matches(InventoryCrafting inv, World worldIn)
+    @Nonnull
+    public NonNullList<ItemStack> getRemainingItems(final InventoryCrafting inventoryCrafting)
     {
-        boolean foundMold = false;
-        for (int slot = 0; slot < inv.getSizeInventory(); slot++)
+        if (Constants.RNG.nextFloat() <= chance)
         {
-            ItemStack stack = inv.getStackInSlot(slot);
-            if (!stack.isEmpty())
+            EntityPlayer player = ForgeHooks.getCraftingPlayer();
+            if (player != null)
             {
-                if (stack.getItem() instanceof ItemMold)
-                {
-                    ItemMold moldItem = ((ItemMold) stack.getItem());
-                    IFluidHandler cap = stack.getCapability(FLUID_HANDLER_CAPABILITY, null);
-
-                    if (cap instanceof IMoldHandler)
-                    {
-                        IMoldHandler moldHandler = ((IMoldHandler) cap);
-                        if (!moldHandler.isMolten())
-                        {
-                            Metal m = moldHandler.getMetal();
-                            if (m != null && moldItem != this.mold && !foundMold)
-                            {
-                                foundMold = true;
-                            }
-                            else
-                            {
-                                return false;
-                            }
-                        }
-                        else
-                        {
-                            return false;
-                        }
-                    }
-                    else
-                    {
-                        return false;
-                    }
-                }
-                else
-                {
-                    return false;
-                }
+                player.addItemStackToInventory(new ItemStack(inventoryCrafting.getStackInSlot(moldSlotID).getItem()));
             }
         }
-        return foundMold;
+
+        return super.getRemainingItems(inventoryCrafting);
     }
 
     @Override
     @Nonnull
-    public ItemStack getCraftingResult(InventoryCrafting inv)
+    public String getGroup()
     {
-        ItemStack moldStack = null;
-        for (int slot = 0; slot < inv.getSizeInventory(); slot++)
+        return group == null ? "" : group.toString();
+    }
+
+    @Override
+    @Nonnull
+    public ItemStack getRecipeOutput() { return ItemStack.EMPTY; }
+
+    @Override
+    public boolean matches(@Nonnull InventoryCrafting inv, @Nonnull World world)
+    {
+        int ingredientCount = 0;
+        RecipeItemHelper recipeItemHelper = new RecipeItemHelper();
```

Comment:
Why did you completely rewrite the logic here? This looks much more complex than it needs to be, and for no particular reason. It's a hardcoded recipe that takes one input, of a particular item, with a particular capability.

Reply:
Nither do I know why I did this, will revert